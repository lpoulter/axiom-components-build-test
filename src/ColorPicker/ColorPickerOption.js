import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';
import './ColorPickerOption.css';

export default class ColorPickerOption extends Component {
  static propTypes = {
    color: PropTypes.oneOf([
      'forbidden-planet',
      'tiny-clanger',
      'critical-mass',
      'fantastic-voyage',
      'paradise-lost',
      'serene-sea',
      'event-horizon',
      'electric-dreams',
      'outer-limits',
      'giant-leap',
      'moon-lagoon',
      'space-invader',
      'extraterrestrial',
      'terra-form',
      'primeval-soup',
      'future-shock',
      'sun-maker',
      'new-horizon',
      'blast-off',
      'crash-course',
      'solar-rust',
      'ground-control',
      'space-oddity',
      'rocky-planet',
      'deep-thought',
      'luna-dust',
    ]),
    disabled: PropTypes.bool,
    onClick: PropTypes.func,
    size: PropTypes.string,
  };

  constructor(props) {
    super(props);
    this.boundHandleClick = this.handleClick.bind(this);
  }

  static contextTypes = {
    closeDropdown: PropTypes.func,
  };

  static defaultProps = {
    disabled: false,
    size: '.75rem',
  };

  handleClick(...args) {
    const { closeDropdown } = this.context;
    const { disabled, onClick } = this.props;

    if (closeDropdown && !disabled) {
      closeDropdown();
    }

    if (onClick) {
      onClick(...args);
    }
  }

  render() {
    const { color = 'empty', disabled, onClick, size, ...rest } = this.props;
    const classes = classnames('ax-color-picker__option', `ax-color-picker__option--${color}`, {
      'ax-color-picker__option--clickable': onClick && !disabled,
      'ax-color-picker__option--disabled': disabled,
    });

    return (
      <Base
          { ...rest }
          className={ classes }
          onClick={ this.boundHandleClick }
          style={ { height: size, width: size } } />
    );
  }
}
