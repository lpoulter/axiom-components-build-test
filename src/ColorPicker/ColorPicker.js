import PropTypes from 'prop-types';
import React, { Component } from 'react';
import atIds from '@brandwatch/axiom-automation-testing/ids';
import Dropdown from '../Dropdown/Dropdown';
import DropdownContent from '../Dropdown/DropdownContent';
import DropdownContext from '../Dropdown/DropdownContext';
import DropdownSource from '../Dropdown/DropdownSource';
import DropdownTarget from '../Dropdown/DropdownTarget';
import Grid from '../Grid/Grid';
import GridCell from '../Grid/GridCell';
import ColorPickerOption from './ColorPickerOption';

export default class ColorPicker extends Component {
  static propTypes = {
    /** Colors that should be disabled from selection */
    disabledOptions: PropTypes.arrayOf(PropTypes.oneOf([
      'forbidden-planet',
      'tiny-clanger',
      'critical-mass',
      'fantastic-voyage',
      'paradise-lost',
      'serene-sea',
      'event-horizon',
      'electric-dreams',
      'outer-limits',
      'giant-leap',
      'moon-lagoon',
      'space-invader',
      'extraterrestrial',
      'terra-form',
      'primeval-soup',
      'future-shock',
      'sun-maker',
      'new-horizon',
      'blast-off',
      'crash-course',
      'solar-rust',
      'ground-control',
      'space-oddity',
      'rocky-planet',
      'deep-thought',
      'luna-dust',
    ])),
    /** Callback for when the color picker has been opened */
    onOpen: PropTypes.func,
    /** Callback for when a color has been selected, provided with the selected color */
    onSelectColor: PropTypes.func.isRequired,
    /** Colors that should be visible for selection */
    options: PropTypes.arrayOf(PropTypes.oneOf([
      'forbidden-planet',
      'tiny-clanger',
      'critical-mass',
      'fantastic-voyage',
      'paradise-lost',
      'serene-sea',
      'event-horizon',
      'electric-dreams',
      'outer-limits',
      'giant-leap',
      'moon-lagoon',
      'space-invader',
      'extraterrestrial',
      'terra-form',
      'primeval-soup',
      'future-shock',
      'sun-maker',
      'new-horizon',
      'blast-off',
      'crash-course',
      'solar-rust',
      'ground-control',
      'space-oddity',
      'rocky-planet',
      'deep-thought',
      'luna-dust',
    ])),
    /** Color that should appear as selected */
    selected: PropTypes.oneOf([
      'forbidden-planet',
      'tiny-clanger',
      'critical-mass',
      'fantastic-voyage',
      'paradise-lost',
      'serene-sea',
      'event-horizon',
      'electric-dreams',
      'outer-limits',
      'giant-leap',
      'moon-lagoon',
      'space-invader',
      'extraterrestrial',
      'terra-form',
      'primeval-soup',
      'future-shock',
      'sun-maker',
      'new-horizon',
      'blast-off',
      'crash-course',
      'solar-rust',
      'ground-control',
      'space-oddity',
      'rocky-planet',
      'deep-thought',
      'luna-dust',
    ]),
    /** Diameter of the color picker button */
    size : PropTypes.string,
    /** Width of the color selection container */
    width: PropTypes.string,
  };

  static defaultProps = {
    disabledOptions: [],
    options: [
      'tiny-clanger',
      'critical-mass',
      'paradise-lost',
      'serene-sea',
      'giant-leap',
      'moon-lagoon',
      'terra-form',
      'primeval-soup',
      'new-horizon',
      'blast-off',
      'ground-control',
      'luna-dust',
    ],
    width: '9rem',
  };

  render() {
    const {
      disabledOptions,
      onOpen,
      onSelectColor,
      options,
      selected,
      width,
      size,
      ...rest
    } = this.props;

    return (
      <Dropdown position="bottom" { ... rest } showArrow>
        <DropdownTarget>
          <ColorPickerOption
              color={ selected }
              data-ax-at={ atIds.ColorPicker.dropdown }
              onClick={ onOpen }
              size={ size } />
        </DropdownTarget>

        <DropdownSource>
          <DropdownContext width={ width }>
            <DropdownContent>
              <Grid
                  gutters="small"
                  horizontalAlign="middle"
                  responsive={ false }>
                { options.map(((color)=> {
                  const isColorDisabled = disabledOptions.indexOf(color) !== -1;
                  return (
                    <GridCell key={ color } shrink>
                      <ColorPickerOption
                          color={ color }
                          data-ax-at={ atIds.ColorPicker.option }
                          disabled={ isColorDisabled }
                          onClick={ () => isColorDisabled || onSelectColor(color) } />
                    </GridCell>
                  );
                } ) ) }
              </Grid>
            </DropdownContent>
          </DropdownContext>
        </DropdownSource>
      </Dropdown>
    );
  }
}
