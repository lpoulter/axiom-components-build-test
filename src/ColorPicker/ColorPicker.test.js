import React from 'react';
import renderer from 'react-test-renderer';
import ColorPicker from './ColorPicker';

const getComponent = (props = {}) =>
  renderer.create(
    <ColorPicker { ...props } onSelectColor={ () => {} } />
  );

describe('ColorPicker', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with a different size', () => {
    const component = getComponent({ size : '3rem' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
