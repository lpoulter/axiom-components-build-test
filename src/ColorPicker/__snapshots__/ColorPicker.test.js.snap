// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker renders with a different size 1`] = `
<div
  className="ax-color-picker__option ax-color-picker__option--empty ax-color-picker__option--clickable"
  data-ax-at="98df24c31c96"
  onClick={[Function]}
  onFocus={[Function]}
  style={
    Object {
      "height": "3rem",
      "width": "3rem",
    }
  }
/>
`;

exports[`ColorPicker renders with defaultProps 1`] = `
<div
  className="ax-color-picker__option ax-color-picker__option--empty ax-color-picker__option--clickable"
  data-ax-at="98df24c31c96"
  onClick={[Function]}
  onFocus={[Function]}
  style={
    Object {
      "height": ".75rem",
      "width": ".75rem",
    }
  }
/>
`;
